lib/theme_mode_handler.dart

```diff
@@ -6,7 +6,7 @@ import 'theme_mode_manager_interface.dart';
 /// to persist and retrieve the user's preference wherever they want.
 class ThemeModeHandler extends StatefulWidget {
   /// Function that runs when themeMode changes.
-  final Widget Function(ThemeMode themeMode) builder;
+  final Widget Function(ThemeMode? themeMode) builder;
```

Comment:
Again I don't think you want this builder to pass back a nullable here. I would guess that by default you get `ThemeMode.system`.

Reply:
I thought that too but then realised that it may be nullable.

When it first loads, `themeMode` is `null` internally, then we load the theme using the `manager.loadThemeMode` and set it.
If `manager.loadThemeMode` returns an invalid value, we set it to `widget.defaultTheme` - (this piece of code](https://github.com/arthurdenner/theme_mode_handler/blob/feat/null-safety/lib/theme_mode_handler.dart#L60-L67)

But if `widget.withFallback` is `true` - the default behaviour - an empty `Container` it's returned - [this piece of code](https://github.com/arthurdenner/theme_mode_handler/blob/feat/null-safety/lib/theme_mode_handler.dart#L71-L73)
So, if `widget.withFallback` is `false`, then `widget.builder` _will_ be called with a `null` value.

Reply:
Thanks for the feedback @kmcgill88!

I thought that too but then realised that it may be nullable.

When it first loads, `themeMode` is `null` internally, then we load the theme using the `manager.loadThemeMode` and set it.
If `manager.loadThemeMode` returns an invalid value, we set it to `widget.defaultTheme` - (this piece of code](https://github.com/arthurdenner/theme_mode_handler/blob/feat/null-safety/lib/theme_mode_handler.dart#L60-L67)

But if `widget.withFallback` is `true` - the default behaviour - an empty `Container` it's returned - [this piece of code](https://github.com/arthurdenner/theme_mode_handler/blob/feat/null-safety/lib/theme_mode_handler.dart#L71-L73)
So, if `widget.withFallback` is `false`, then `widget.builder` _will_ be called with a `null` value.

Reply:
I tried to use the `late` keyword here but it'd mean that `themeMode` is not `null` - which isn't true in the last scenario I described above.

What are your thoughts?

Reply:
hmmm I see what you mean. Could you refactor [build](https://github.com/arthurdenner/theme_mode_handler/blob/feat/null-safety/lib/theme_mode_handler.dart#L70) to use a `FutureBuilder` with the Future from loading the theme from disk? You could remove `withFallback` and replace that with a `paceHolderWidget` to display while the theme loads. This way users can define their own behavior while the theme gets loaded from disk.

If you went this direction you could make the theme non-nullable b/c the builder wouldn't be called unless there was a theme. If no theme on disk, just use `defaultTheme` that you currently have.

Reply:
I think I'm biased to the theme passed via builder is non-nullable but then again users can easily take the builder result and do a default value themselves. 🤷🏻‍♂️

Reply:
I really like your idea. It gives more control to the users too, which I prefer. I'll work on it. Thanks!